nf_core/components/components_install.py

```diff
@@ -0,0 +1,127 @@
+        return False
+
+    if sha:
+        if not modules_repo.sha_exists_on_branch(sha):
+            log.error(f"Commit SHA '{sha}' doesn't exist in '{modules_repo.remote_url}'")
+            return False
+
+    return True
+
+
+def collect_and_verify_name(component_type, component, modules_repo):
+    """
+    Collect component name.
+    Check that the supplied name is an available module/subworkflow.
+    """
+    if component is None:
+        component = questionary.autocomplete(
+            f"{'Tool' if component_type == 'modules' else 'Subworkflow'} name:",
+            choices=modules_repo.get_avail_components(component_type),
+            style=nf_core.utils.nfcore_question_style,
+        ).unsafe_ask()
+
+    # Check that the supplied name is an available module/subworkflow
+    if component and component not in modules_repo.get_avail_components(component_type):
+        log.error(f"{component_type[:-1].title()} '{component}' not found in list of available {component_type}.")
+        log.info(f"Use the command 'nf-core {component_type} list' to view available software")
+        return False
+
+    if not modules_repo.component_exists(component, component_type):
+        warn_msg = f"{component_type[:-1].title()} '{component}' not found in remote '{modules_repo.remote_url}' ({modules_repo.branch})"
+        log.warning(warn_msg)
+        return False
+
+    return component
+
+
+def check_component_installed(component_type, component, current_version, component_dir, modules_repo, force):
+    """
+    Check that the module/subworkflow is not already installed
+    """
+    if (current_version is not None and os.path.exists(component_dir)) and not force:
+        log.info(f"{component_type[:-1].title()} is already installed.")
+
+        message = "?" if component_type == "modules" else " of this subworkflow and all it's imported modules?"
+        force = questionary.confirm(
+            f"{component_type[:-1].title()} {component} is already installed. \nDo you want to force the reinstallation{message}",
+            style=nf_core.utils.nfcore_question_style,
+            default=False,
+        ).unsafe_ask()
+
+        if not force:
+            repo_flag = "" if modules_repo.repo_path == NF_CORE_MODULES_NAME else f"-g {modules_repo.remote_url} "
+            branch_flag = "" if modules_repo.branch == "master" else f"-b {modules_repo.branch} "
+
+            log.info(
+                f"To update '{component}' run 'nf-core {component_type} {repo_flag}{branch_flag}update {component}'. To force reinstallation use '--force'"
+            )
+            return False
+
+    return True
+
+
+def get_version(component, component_type, sha, prompt, current_version, modules_repo):
+    """
+    Get the version to install
+    """
+    if sha:
+        version = sha
+    elif prompt:
+        try:
+            version = nf_core.modules.modules_utils.prompt_module_version_sha(
```

Comment:
should we move `prompt_module_version_sha` to component_utils and make it `prompt_component_version_sha`?

Reply:
done in [23c457c](https://github.com/nf-core/tools/pull/1980/commits/23c457ce80465911bb28ad44568574ba9c6cf726)